contracts/exchange/Orderbook.sol

```diff
@@ -139,6 +144,40 @@ contract Orderbook is IOrderbook, ManagerBase {
                 orderAmounts[i] = orders[_orderIds[i]].amountOrdered - orders[_orderIds[i]].amountFilled;
             }
         }
+
+        // get amounts ordered based on AmountToFill and Max Spend
+        amountFilled = 0;
+        uint256 amountSpentOnOrder = 0;
+        for (uint256 i = 0; i < orderAmounts.length; ++i) {
+            if (orderAmounts[i] > 0) {
+                amountSpentOnOrder = orders[_orderIds[i]].price * orderAmounts[i];
+                
+                // Check if there's you're still under tha max Spend
+                if (maxSpend == 0) {
+                    orderAmounts[i] = 0;
+                    continue;
+                } else if (maxSpend > amountSpentOnOrder) {
+                    maxSpend -= amountSpentOnOrder;
+                } else if (maxSpend < amountSpentOnOrder) {
+                    orderAmounts[i] = maxSpend / orders[_orderIds[i]].price;
+                    maxSpend = 0;
+                }
+
+                if (orderAmounts[i] <= amountToFill) {
```

Comment:
What if orderAmounts[i] == 0 and amountToFill == 0? You would get into this first if statement. If that's not desired, I'd add a "&& amountToFill > 0) check to the if.

Reply:
both being equal to zero doesn't really matter here since it'll subtract 0 from amountToFill and add 0 to amountFilled. 

But I get what you're saying, it'll be a waste of gas. I'll do a 'break' to exit immediately if amountToFill is already at 0 or maxSpend is already at 0.

Reply:
Ignore my last statement. 

This is fine. if orderamounts[i] == 0 and amountToFill == 0 will only occur if maxSpend == 0 (which exists early) or maxSpend is less than amountSpendOnOrder. If it's the second case, orderAmount[i] is set to 0 and it'll only hit this ONCE. This is expected and a non-issue. There's no getting around it. it'll subtract 0 from amountToFill and add 0 to amountFilled so it'll be a no-op anyway.